### ISSUE TITLE
Works with Lenovo ThinkPad X390 under Debian 10 - except automatic DNS setup

### ISSUE THREAD

**Issue by DDvO** [xmm7360/xmm7360-pci#12]
Thanks @abrasive for this new solution - very cool!
It works much better for me than https://github.com/xmm7360/xmm7360-usb-modeswitch
Praying before running `sudo python3 rpc/open_xdatachannel.py` the first time apparently did help :-)

As already mentioned by another user in #7, I also need to set up DNS manually.
I do get some direct routing - `ip route`  yields 
```
default dev wwan0 proto static
```
From two of the many lines of output of the Python script (in my case for `web.vodafone.de`):
 ```
(IPv4Address('0.0.0.82'), IPv4Address('31.213.168.1'), IPv4Address('100.98.7.178'))
{'v4': [IPv4Address('139.7.30.125'), IPv4Address('139.7.30.126'), IPv4Address('139.7.30.125')], 'v6': [IPv6Address('2a01:860:0:300::fffe:53'), IPv6Address('2a01:860:0:300::ffff:53')]}
```
I can copy and use one of the DNS server IP addresses. For instance:
```
echo 'nameserver 139.7.30.126' > /etc/resolv.conf
```
but I wonder why this is not done somehow automatically?

**Comment by abrasive** [comment]
It's not done automatically because managing `resolv.conf` is a Hard Problem eg. putting it back when we're "done". Really this needs to be handled by a daemon like `ModemManager`. I guess we could append them to `resolv.conf` for now?

**Comment by abrasive** [comment]
Fixed in 7b12f3f8024e827d4f6b622cb93ba1051625212d (and later made configurable to taste)

**Comment by DDvO** [comment]
I agree that appending to ` /etc/resolv.conf` is at least a viable interim solution.
Just please don't add the same entires again and again (when the connection is closed and later re-opened).
Checking for the existence of `# Added by xmm7360` and not adding entries if present would be a simple possibility, but this would not support changing DNS entries.
A more clever possibility would be: for each DNS entry to be added, check it it is already there and not add it again if so.
